build/templates/session.py.mako

```diff
@@ -32,77 +32,70 @@ from ${module_name} import library_singleton
 
     def __get__(self, obj, objtype):
         assert objtype is Session
-        return obj._get_attribute_vi_int32(self.channel, self.attribute_id)
+        return self.get(obj, self._default_channel)
 
     def __set__(self, obj, value):
-        obj._set_attribute_vi_int32(self.channel, self.attribute_id, value)
+        return self.set(obj, self._default_channel, value)
 
 
-class AttributeViReal64(object):
+class AttributeViInt32(Attribute):
 
-    def __init__(self, attribute_id):
-        self.attribute_id = attribute_id
-        self.channel = ''
+    def get(self, session, channel):
+        return session._get_attribute_vi_int32(channel, self._attribute_id)
 
-    def __get__(self, obj, objtype):
-        assert objtype is Session
-        return obj._get_attribute_vi_real64(self.channel, self.attribute_id)
+    def set(self, session, channel, value):
+        session._set_attribute_vi_int32(channel, self._attribute_id, value)
 
-    def __set__(self, obj, value):
-        obj._set_attribute_vi_real64(self.channel, self.attribute_id, value)
 
+class AttributeViReal64(Attribute):
 
-class AttributeViString(object):
+    def get(self, session, channel):
+        return session._get_attribute_vi_real64(channel, self._attribute_id)
 
-    def __init__(self, attribute_id):
-        self.attribute_id = attribute_id
-        self.channel = ''
+    def set(self, session, channel, value):
+        session._set_attribute_vi_real64(channel, self._attribute_id, value)
 
-    def __get__(self, obj, objtype):
-        assert objtype is Session
-        return obj._get_attribute_vi_string(self.channel, self.attribute_id)
 
-    def __set__(self, obj, value):
-        obj._set_attribute_vi_string(self.channel, self.attribute_id, value)
+class AttributeViString(Attribute):
 
+    def get(self, session, channel):
+        return session._get_attribute_vi_string(channel, self._attribute_id)
 
-class AttributeViBoolean(object):
+    def set(self, session, channel, value):
+        session._set_attribute_vi_string(channel, self._attribute_id, value)
 
-    def __init__(self, attribute_id):
-        self.attribute_id = attribute_id
-        self.channel = ''
 
-    def __get__(self, obj, objtype):
-        assert objtype is Session
-        return obj._get_attribute_vi_boolean(self.channel, self.attribute_id)
+class AttributeViBoolean(Attribute):
 
-    def __set__(self, obj, value):
-        obj._set_attribute_vi_boolean(self.channel, self.attribute_id, value)
+    def get(self, session, channel):
+        return session._get_attribute_vi_boolean(channel, self._attribute_id)
 
+    def set(self, session, channel, value):
+        session._set_attribute_vi_boolean(channel, self._attribute_id, value)
 
-class AttributeEnum(object):
 
-    def __init__(self, attribute_id, enum_meta_class):
-        self.attribute_id = attribute_id
-        self.attribute_type = enum_meta_class
-        self.channel = ''
+class AttributeEnum(AttributeViInt32):
```

Comment:
I don't think you want to inherit from ViInt32. I believe Scope will have some float enums. Specifically for input impedance.

Or Maybe name it AttributeEnumViInt32?

Reply:
Let's fix it when we need to so we can test exactly the right scenario. Is that ok?